openmetadata-service/src/main/java/org/openmetadata/service/resources/dqtests/TestSuiteResource.java

```diff
@@ -469,9 +472,64 @@ public TestSummary getTestsExecutionSummary(
     OperationContext operationContext =
         new OperationContext(Entity.TABLE, MetadataOperation.VIEW_TESTS);
     authorizer.authorize(securityContext, operationContext, resourceContext);
+    // Set the deprecation header based on draft specification from IETF
+    // https://datatracker.ietf.org/doc/html/draft-ietf-httpapi-deprecation-header-02
+    response.setHeader("Deprecation", "Monday, October 30, 2024");
+    response.setHeader(
+        "Link", "api/v1/dataQuality/testSuites/dataQualityReport; rel=\"alternate\"");
     return repository.getTestSummary(testSuiteId);
   }
 
+  @GET
+  @Path("/dataQualityReport")
+  @Operation(
+      operationId = "getDataQualityReport",
+      summary = "Get Data Quality Report",
+      description =
+          """
+            Use the search service to perform data quality aggregation. You can use the `q` parameter to filter the results.
+            the `aggregationQuery` is of the form `bucketName=<bucketName>:aggType=<aggType>:field=<field>`. You can sperate aggregation
+            query with a comma `,` to perform nested aggregations.
+            For example, `bucketName=table:aggType=terms:field=databaseName,bucketName=<bucketName>:aggType=<aggType>:field=<field>`
+            """,
+      responses = {
+        @ApiResponse(
+            responseCode = "200",
+            description = "Data Quality Report Results",
+            content =
+                @Content(
+                    mediaType = "application/json",
+                    schema = @Schema(implementation = DataQualityReport.class)))
+      })
+  public DataQualityReport getDataQualityReport(
+      @Context UriInfo uriInfo,
+      @Context SecurityContext securityContext,
+      @Parameter(
+              description = "Search query to filter the aggregation results",
+              schema = @Schema(type = "String"))
+          @QueryParam("q")
+          String query,
+      @Parameter(
+              description = "Aggregation query to perform aggregation on the search results",
+              schema = @Schema(type = "String"))
+          @QueryParam("aggregationQuery")
+          String aggregationQuery,
+      @Parameter(
+              description = "Index to perform the aggregation against",
+              schema = @Schema(type = "String"))
+          @QueryParam("index")
+          String index)
+      throws IOException {
+    ResourceContext<?> resourceContext = getResourceContext();
+    OperationContext operationContext =
+        new OperationContext(entityType, MetadataOperation.VIEW_TESTS);
+    authorizer.authorize(securityContext, operationContext, resourceContext);
+    if (nullOrEmpty(aggregationQuery) || nullOrEmpty(index)) {
+      throw new IllegalArgumentException("aggregationQuery and index are required parameters");
+    }
+    return repository.getDataQualityReport(query, aggregationQuery, index);
```

Comment:
In future changes, I suggest looking into leveraging the schema to validate the query. Currently any query or fields can be passed here with not validation.